filelog_windows.go

```diff
@@ -1,3 +1,76 @@
 package log4go
 
+import (
+	"archive/zip"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
 const FILELOG_ARCHIVE_REGEX = `\.[0-9]{4}-[0-9]{2}-[0-9]{2}(\.[0-9]{4})?(\.zip)?$`
+
+func (w *FileLogWriter) compressFile(filename string) {
```

Comment:
`gz` is straight compression, but `zip` is a container format that has an additional level of nesting (see line 32 for the creation of the zip file entry).